pkg/controller/cache/cache_controller.go

```diff
@@ -92,10 +111,163 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
+	// Reconcile cache
 	reqLogger.Info("Identify the target cluster")
-	reqLogger.Info("If cache doesn't exist create")
-	reqLogger.Info("If cache already exists, check if it's compatible with the CR request. If not then do appropriate actions")
-	reqLogger.Info("Update CR status with connnection info")
+	// Fetch the Infinispan cluster info
+	ispnInstance := &infinispanv1.Infinispan{}
+	nsName := types.NamespacedName{Name: instance.Spec.ClusterName, Namespace: instance.Namespace}
+	err = r.client.Get(context.TODO(), nsName, ispnInstance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Error(err, "Infinispan cluster "+ispnInstance.Name+" not found")
+			return reconcile.Result{RequeueAfter: time.Second * 10}, err
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Cluster must be well formed
+	for i, c := range ispnInstance.Status.Conditions {
```

Comment:
@rigazilla what do you think if we will create IsWellFormed method during refactoring on v1.Infinispan type?

Reply:
sounds good

Reply:
@rigazilla have you decide to create a IsWellFormed function for this?

Reply:
yeah,  Let me add it. (I though it was in your refactoring plans.)

Reply:
@dmvolod done